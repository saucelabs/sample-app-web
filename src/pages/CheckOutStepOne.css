.checkout_info_container {
  max-width: 1280px;
  margin: 0 auto;
  padding-top: 30px;
}

.checkout_info_wrapper {
  padding: 0 15px;
}

.checkout_info {
  max-width: 380px;
  margin: 0 auto;
  padding-top: 30px;
}

.checkout_buttons {
  margin-top: 5px;
  border-top: 2px solid #ededef;
  display: flex;
  justify-content: space-between;
  padding: 40px 0;
  margin-bottom: 30px;
}

.checkout_buttons .btn {
  width: 220px;
}

.checkout_buttons .btn {
  margin-bottom: 0;
}

@media only screen and (min-width: 900px) {
  .checkout_info {
    border: 2px solid #ededef;
    padding: 40px 40px 0 40px;
    margin: 70px auto 100px auto;
  }
}

@media only screen and (max-width: 640px) {
  .checkout_buttons {
    display: block;
    justify-content: inherit;
  }

  .checkout_buttons .btn {
    margin-bottom: 15px;
    width: 100%;
  }
}
