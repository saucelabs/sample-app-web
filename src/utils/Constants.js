export const FONT_FAMILY = "Roboto, Arial, Helvetica, sans-serif";
export const VALID_USERNAMES = [
  "standard_user",
  "locked_out_user",
  "problem_user",
  "performance_glitch_user",
  "error_user",
  "visual_user",
];
export const VALID_PASSWORD = "secret_sauce";
export const ROUTES = {
  LOGIN: "/",
  INVENTORY: "/inventory.html",
  INVENTORY_LIST: "/inventory-item.html",
  CART: "/cart.html",
  CHECKOUT_STEP_ONE: "/checkout-step-one.html",
  CHECKOUT_STEP_TWO: "/checkout-step-two.html",
  CHECKOUT_COMPLETE: "/checkout-complete.html",
};
export const CART_CONTENTS = "cart-contents";
export const SESSION_USERNAME = "session-username";
