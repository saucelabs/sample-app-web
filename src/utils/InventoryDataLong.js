export const InventoryDataLong = [
  {
    id: 0,
    name: "Sauce Labs Bike Light",
    desc: `A red light isn't the desired state in testing but it sure helps when riding your bike at night. Water-resistant with 3 lighting modes, 1 AAA battery included.`,
    price: 9.99,
    image_url: "bike-light-1200x1500.jpg",
  },
  {
    id: 1,
    name: "Sauce Labs Bolt T-Shirt (XS)",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
    image_url: "bolt-shirt-1200x1500.jpg",
  },
  {
    id: 2,
    name: "Sauce Labs Onesie",
    desc: `Rib snap infant onesie for the junior automation engineer in development. Reinforced 3-snap bottom closure, two-needle hemmed sleeved and bottom won't unravel.`,
    price: 7.99,
    image_url: "red-onesie-1200x1500.jpg",
  },
  {
    id: 3,
    name: "Test.allTheThings() T-Shirt (Red) (XS)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },
  {
    id: 4,
    name: "Sauce Labs Backpack",
    desc: `carry.allTheThings() with the sleek, streamlined Sly Pack that melds uncompromising style with unequaled laptop and tablet protection.`,
    price: 29.99,
    image_url: "sauce-backpack-1200x1500.jpg",
  },
  {
    id: 5,
    name: "Sauce Labs Fleece Jacket (XS)",
    desc: `It's not every day that you come across a midweight quarter-zip fleece jacket capable of handling everything from a relaxing day outdoors to a busy day at the office.`,
    price: 49.99,
    image_url: "sauce-pullover-1200x1500.jpg",
  },
  {
    id: 6,
    name: "Sauce Labs Fleece Jacket (S)",
    desc: `It's not every day that you come across a midweight quarter-zip fleece jacket capable of handling everything from a relaxing day outdoors to a busy day at the office.`,
    price: 49.99,
    image_url: "sauce-pullover-1200x1500.jpg",
  },
  {
    id: 7,
    name: "Sauce Labs Fleece Jacket (M)",
    desc: `It's not every day that you come across a midweight quarter-zip fleece jacket capable of handling everything from a relaxing day outdoors to a busy day at the office.`,
    price: 49.99,
    image_url: "sauce-pullover-1200x1500.jpg",
  },
  {
    id: 8,
    name: "Sauce Labs Fleece Jacket (L)",
    desc: `It's not every day that you come across a midweight quarter-zip fleece jacket capable of handling everything from a relaxing day outdoors to a busy day at the office.`,
    price: 49.99,
    image_url: "sauce-pullover-1200x1500.jpg",
  },
  {
    id: 9,
    name: "Sauce Labs Fleece Jacket (XL)",
    desc: `It's not every day that you come across a midweight quarter-zip fleece jacket capable of handling everything from a relaxing day outdoors to a busy day at the office.`,
    price: 49.99,
    image_url: "sauce-pullover-1200x1500.jpg",
  },
  {
    id: 10,
    name: "Sauce Labs Fleece Jacket (XXL)",
    desc: `It's not every day that you come across a midweight quarter-zip fleece jacket capable of handling everything from a relaxing day outdoors to a busy day at the office.`,
    price: 49.99,
    image_url: "sauce-pullover-1200x1500.jpg",
  },
  {
    id: 11,
    name: "Test.allTheThings() T-Shirt (Red) (S)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },

  {
    id: 12,
    name: "Test.allTheThings() T-Shirt (Red) (M)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },

  {
    id: 13,
    name: "Test.allTheThings() T-Shirt (Red) (L)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },

  {
    id: 14,
    name: "Test.allTheThings() T-Shirt (Red) (L)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },

  {
    id: 15,
    name: "Test.allTheThings() T-Shirt (Red) (XL)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },

  {
    id: 16,
    name: "Test.allTheThings() T-Shirt (Red) (XXL)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },
  {
    id: 17,
    name: "Sauce Labs Bolt T-Shirt (M)",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
    image_url: "bolt-shirt-1200x1500.jpg",
  },
  {
    id: 18,
    name: "Sauce Labs Bolt T-Shirt (L)",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
    image_url: "bolt-shirt-1200x1500.jpg",
  },
  {
    id: 19,
    name: "Sauce Labs Bolt T-Shirt (XL)",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
    image_url: "bolt-shirt-1200x1500.jpg",
  },
  {
    id: 20,
    name: "Sauce Labs Bolt T-Shirt (XXL)",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
    image_url: "bolt-shirt-1200x1500.jpg",
  },
  {
    id: 21,
    name: "Sauce Labs Bolt T-Shirt (S)",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
    image_url: "bolt-shirt-1200x1500.jpg",
  },
];
