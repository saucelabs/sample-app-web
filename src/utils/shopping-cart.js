export class ShoppingCart {
  static addItem(itemId) {
    // pull out our current cart contents
    const curContents = ShoppingCart.getCartContents();

    /* istanbul ignore else */
    if (curContents.indexOf(itemId) < 0) {
      // Item's not yet present - add it now
      curContents.push(itemId);

      // We modified our cart, so store it now
      ShoppingCart.setCartContents(curContents);
    }
  }

  static removeItem(itemId) {
    // pull out our current cart contents
    const curContents = ShoppingCart.getCartContents();
    const itemIndex = curContents.indexOf(itemId);

    /* istanbul ignore else */
    if (itemIndex >= 0) {
      // Remove this item from the array
      curContents.splice(itemIndex, 1);

      // We modified our cart, so store it now
      ShoppingCart.setCartContents(curContents);
    }
  }

  static isItemInCart(itemId) {
    // pull out our current cart contents
    const curContents = ShoppingCart.getCartContents();

    // If the item is in the array, return true
    return curContents.indexOf(itemId) >= 0;
  }

  static getCartContents() {
    // pull out our current cart contents
    let curContents = window.localStorage.getItem("cart-contents");

    // Make an empty list if this is the first item
    if (curContents == null) {
      curContents = [];
    } else {
      // We have an existing cart, so deserialize it now since localStorage stores in JSON strings
      curContents = JSON.parse(curContents);
    }

    return curContents;
  }

  static setCartContents(newContents) {
    window.localStorage.setItem("cart-contents", JSON.stringify(newContents));

    // Notify our listeners
    /* istanbul ignore next */
    ShoppingCart.LISTENERS.forEach((curListener) => {
      curListener.forceUpdate();
    });
  }

  static resetCart() {
    window.localStorage.removeItem("cart-contents");

    // Notify our listeners
    /* istanbul ignore next */
    ShoppingCart.LISTENERS.forEach((curListener) => {
      curListener.forceUpdate();
    });
  }

  static registerCartListener(handler) {
    ShoppingCart.LISTENERS.push(handler);
  }
}

ShoppingCart.LISTENERS = [];
