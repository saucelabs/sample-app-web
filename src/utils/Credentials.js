import Cookies from "js-cookie";
import { SESSION_USERNAME, VALID_PASSWORD, VALID_USERNAMES } from "./Constants";

/**
 * Verify the credentials
 *
 * @param {string} username
 * @param {string} password
 *
 * @return {boolean}
 */
export function verifyCredentials(username, password) {
  if (password !== VALID_PASSWORD) {
    return false;
  }

  return VALID_USERNAMES.includes(username);
}

/**
 * Store the data in our cookies
 *
 * @param {string} username
 *
 * @param {string} password
 */
export function setCredentials(username, password) {
  let date = new Date();
  date.setTime(date.getTime() + 10 * 60 * 1000);

  Cookies.set(SESSION_USERNAME, username, { expires: date });
}

/**
 * Remove the credentials
 */
export function removeCredentials() {
  Cookies.remove(SESSION_USERNAME);
}

/**
 * Return current logged username
 *
 * @return {string | undefined}
 */
export function currentUser() {
  return Cookies.get(SESSION_USERNAME);
}

/**
 * Check if this is a problem user
 *
 * @return {boolean}
 */
export function isProblemUser() {
  return Cookies.get(SESSION_USERNAME) === "problem_user";
}

/**
 * Check if this is a performance user
 *
 * @return {boolean}
 */
export function isPerformanceGlitchUser() {
  return Cookies.get(SESSION_USERNAME) === "performance_glitch_user";
}

/**
 * Check if this a logged out user
 *
 * @return {boolean}
 */
export function isLockedOutUser() {
  return Cookies.get(SESSION_USERNAME) === "locked_out_user";
}

/**
 * Check if this is an error user
 *
 * @return {boolean}
 */
export function isErrorUser() {
  return Cookies.get(SESSION_USERNAME) === "error_user";
}

/**
 * Check if the user is logged in with a valid username
 *
 * @return {boolean}
 */
export function isLoggedIn() {
  const sessionUsername = Cookies.get(SESSION_USERNAME);
  const isValidUsername = VALID_USERNAMES.includes(sessionUsername);

  return isValidUsername && sessionUsername !== "locked_out_user";
}

/**
 * Check if this is a visual user
 *
 * @return {boolean}
 */
export function isVisualUser() {
  return Cookies.get(SESSION_USERNAME) === "visual_user";
}
