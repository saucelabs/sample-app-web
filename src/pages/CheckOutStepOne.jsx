import React, { useState } from "react";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";
import { isProblemUser } from "../utils/Credentials";
import { ROUTES } from "../utils/Constants";
import SwagLabsFooter from "../components/Footer";
import HeaderContainer from "../components/HeaderContainer";
import InputError, { INPUT_TYPES } from "../components/InputError";
import ErrorMessage from "../components/ErrorMessage";
import SubmitButton from "../components/SubmitButton";
import Button, { BUTTON_SIZES, BUTTON_TYPES } from "../components/Button";
import "./CheckOutStepOne.css";

const CheckOutStepOne = ({ history }) => {
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [postalCode, setPostalCode] = useState("");
  const [error, setError] = useState("");
  const dismissError = () => {
    setError("");
  };
  const handleFirstNameChange = (evt) => {
    setFirstName(evt.target.value);
  };
  const handleLastNameChange = (evt) => {
    if (isProblemUser()) {
      // Overwrite the firstname also
      return setFirstName(evt.target.value);
    }

    setLastName(evt.target.value);
  };
  const handlePostalCodeChange = (evt) => {
    setPostalCode(evt.target.value);
  };
  const handleSubmit = (evt) => {
    evt.preventDefault();

    if (!firstName) {
      return setError("First Name is required");
    }

    if (!lastName) {
      return setError("Last Name is required");
    }

    if (!postalCode) {
      return setError("Postal Code is required");
    }

    // If we're here, we have our required info. Redirect!
    history.push(ROUTES.CHECKOUT_STEP_TWO);

    return "";
  };

  return (
    <div id="page_wrapper" className="page_wrapper">
      <div id="contents_wrapper">
        <HeaderContainer secondaryTitle="Checkout: Your Information" />
        <div id="checkout_info_container" className="checkout_info_container">
          <div className="checkout_info_wrapper">
            <form onSubmit={handleSubmit}>
              <div className="checkout_info">
                <InputError
                  isError={Boolean(error)}
                  type={INPUT_TYPES.TEXT}
                  value={firstName}
                  onChange={handleFirstNameChange}
                  testId="firstName"
                  placeholder="First Name"
                  // Custom
                  id="first-name"
                  autoCorrect="off"
                  autoCapitalize="none"
                />
                <InputError
                  isError={Boolean(error)}
                  type={INPUT_TYPES.TEXT}
                  value={lastName}
                  onChange={handleLastNameChange}
                  testId="lastName"
                  placeholder="Last Name"
                  // Custom
                  id="last-name"
                  autoCorrect="off"
                  autoCapitalize="none"
                />
                <InputError
                  isError={Boolean(error)}
                  type={INPUT_TYPES.TEXT}
                  value={postalCode}
                  onChange={handlePostalCodeChange}
                  testId="postalCode"
                  placeholder="Zip/Postal Code"
                  // Custom
                  id="postal-code"
                  autoCorrect="off"
                  autoCapitalize="none"
                />
                <ErrorMessage
                  isError={Boolean(error)}
                  errorMessage={`Error: ${error}`}
                  onClick={dismissError}
                />
              </div>
              <div className="checkout_buttons">
                <Button
                  // `cart_cancel_link` has no style function
                  // but is there for backwards compatibility
                  customClass="cart_cancel_link"
                  label="Cancel"
                  onClick={(evt) => {
                    evt.preventDefault();
                    history.push(ROUTES.CART);
                  }}
                  size={BUTTON_SIZES.MEDIUM}
                  testId="cancel"
                  type={BUTTON_TYPES.BACK}
                />
                <SubmitButton
                  customClass="btn btn_primary cart_button btn_action"
                  testId="continue"
                  value="Continue"
                />
              </div>
            </form>
          </div>
        </div>
      </div>
      <SwagLabsFooter />
    </div>
  );
};
CheckOutStepOne.propTypes = {
  /**
   * The history
   */
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export default withRouter(CheckOutStepOne);
