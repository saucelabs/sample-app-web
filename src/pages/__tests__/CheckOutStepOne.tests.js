import React, { useState as useStateMock } from "react";
import { shallow } from "enzyme";
import CheckOutStepOne from "../CheckOutStepOne";
import * as Credentials from "../../utils/Credentials";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: jest.fn(),
  useEffect: (f) => f(),
}));

let props;

describe("CheckOutStepOne", () => {
  const setState = jest.fn();

  beforeEach(() => {
    props = {
      history: {
        push: jest.fn(),
      },
    };
    useStateMock.mockImplementation((init) => [init, setState]);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    const wrapper = shallow(<CheckOutStepOne.WrappedComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should be able to dismiss the error", () => {
    const wrapper = shallow(<CheckOutStepOne.WrappedComponent {...props} />);
    const ErrorMessage = wrapper.find("ErrorMessage");

    ErrorMessage.at(0).simulate("click");

    expect(setState).toHaveBeenCalledWith("");
  });

  it("should update the first name state when data is typed in the first name field", () => {
    const wrapper = shallow(<CheckOutStepOne.WrappedComponent {...props} />);
    const InputErrors = wrapper.find("InputError");
    const value = "first name";

    InputErrors.at(0).simulate("change", { target: { value } });

    expect(setState).toHaveBeenCalledWith(value);
  });

  it("should update the last name state when data is typed in the last name field", () => {
    const wrapper = shallow(<CheckOutStepOne.WrappedComponent {...props} />);
    const InputErrors = wrapper.find("InputError");
    const value = "last name";

    InputErrors.at(1).simulate("change", { target: { value } });

    expect(setState).toHaveBeenCalledWith(value);
  });

  it("should update the last name state when data is typed in the last name field for a problem user", () => {
    const isProblemUserSpy = jest.spyOn(Credentials, "isProblemUser");
    isProblemUserSpy.mockReturnValueOnce(true);
    const wrapper = shallow(<CheckOutStepOne.WrappedComponent {...props} />);
    const InputErrors = wrapper.find("InputError");
    const value = "last name";

    InputErrors.at(1).simulate("change", { target: { value } });

    expect(isProblemUserSpy).toHaveBeenCalledTimes(1);
    expect(setState).toHaveBeenCalledWith(value);
  });

  it("should update the zip code state when data is typed in the zip code field", () => {
    const wrapper = shallow(<CheckOutStepOne.WrappedComponent {...props} />);
    const InputErrors = wrapper.find("InputError");
    const value = "zip code";

    InputErrors.at(2).simulate("change", { target: { value } });

    expect(setState).toHaveBeenCalledWith(value);
  });

  it("should set the no first name state when the checkout button is being pressed", () => {
    const wrapper = shallow(<CheckOutStepOne.WrappedComponent {...props} />);
    const form = wrapper.find("form").at(0);
    form.simulate("submit", {
      preventDefault() {},
    });

    expect(setState).toHaveBeenCalledWith("First Name is required");
  });

  it("should set the no last name state when the checkout button is being pressed", () => {
    useStateMock
      // first name useState
      .mockImplementationOnce((stateValue) => [
        (stateValue = "first name"),
        setState,
      ]);

    const wrapper = shallow(<CheckOutStepOne.WrappedComponent {...props} />);
    const InputErrors = wrapper.find("InputError");
    InputErrors.at(0).simulate("change", { target: { value: "first name" } });
    const form = wrapper.find("form").at(0);
    form.simulate("submit", {
      preventDefault() {},
    });

    expect(setState).toHaveBeenCalledTimes(2);
    expect(setState).toHaveBeenNthCalledWith(2, "Last Name is required");
  });

  it("should set the no zip code state when the checkout button is being pressed", () => {
    useStateMock
      // first name useState
      .mockImplementationOnce((stateValue) => [
        (stateValue = "first name"),
        setState,
      ])
      // last name useState
      .mockImplementationOnce((stateValue) => [
        (stateValue = "last name"),
        setState,
      ]);

    const wrapper = shallow(<CheckOutStepOne.WrappedComponent {...props} />);
    const InputErrors = wrapper.find("InputError");
    InputErrors.at(0).simulate("change", { target: { value: "first name" } });
    InputErrors.at(1).simulate("change", { target: { value: "last name" } });
    const form = wrapper.find("form").at(0);
    form.simulate("submit", {
      preventDefault() {},
    });

    expect(setState).toHaveBeenCalledTimes(3);
    expect(setState).toHaveBeenNthCalledWith(3, "Postal Code is required");
  });

  it("should redirect to the checkout when valid date is used", () => {
    useStateMock
      // first name useState
      .mockImplementationOnce((stateValue) => [
        (stateValue = "first name"),
        setState,
      ])
      // last name useState
      .mockImplementationOnce((stateValue) => [
        (stateValue = "last name"),
        setState,
      ])
      // zip code useState
      .mockImplementationOnce((stateValue) => [
        (stateValue = "zip code"),
        setState,
      ]);

    const wrapper = shallow(<CheckOutStepOne.WrappedComponent {...props} />);
    const InputErrors = wrapper.find("InputError");
    InputErrors.at(0).simulate("change", { target: { value: "first name" } });
    InputErrors.at(1).simulate("change", { target: { value: "last name" } });
    InputErrors.at(2).simulate("change", { target: { value: "zip code" } });
    const form = wrapper.find("form").at(0);
    form.simulate("submit", {
      preventDefault() {},
    });

    expect(setState).toHaveBeenCalledTimes(3);
    expect(props.history.push).toBeCalledWith("/checkout-step-two.html");
  });

  it("should redirect to the cart when the cancel button is clicked", () => {
    const wrapper = shallow(<CheckOutStepOne.WrappedComponent {...props} />);
    const backButton = wrapper.find("Button").at(0);
    backButton.simulate("click", {
      preventDefault() {},
    });

    expect(props.history.push).toBeCalledWith("/cart.html");
  });
});
