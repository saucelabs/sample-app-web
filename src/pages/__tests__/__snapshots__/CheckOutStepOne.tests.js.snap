// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckOutStepOne should render correctly 1`] = `
<div
  className="page_wrapper"
  id="page_wrapper"
>
  <div
    id="contents_wrapper"
  >
    <HeaderContainer
      secondaryTitle="Checkout: Your Information"
    />
    <div
      className="checkout_info_container"
      id="checkout_info_container"
    >
      <div
        className="checkout_info_wrapper"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="checkout_info"
          >
            <InputError
              autoCapitalize="none"
              autoCorrect="off"
              id="first-name"
              isError={false}
              onChange={[Function]}
              placeholder="First Name"
              testId="firstName"
              type="text"
              value=""
            />
            <InputError
              autoCapitalize="none"
              autoCorrect="off"
              id="last-name"
              isError={false}
              onChange={[Function]}
              placeholder="Last Name"
              testId="lastName"
              type="text"
              value=""
            />
            <InputError
              autoCapitalize="none"
              autoCorrect="off"
              id="postal-code"
              isError={false}
              onChange={[Function]}
              placeholder="Zip/Postal Code"
              testId="postalCode"
              type="text"
              value=""
            />
            <ErrorMessage
              errorMessage="Error: "
              isError={false}
              onClick={[Function]}
            />
          </div>
          <div
            className="checkout_buttons"
          >
            <Button
              customClass="cart_cancel_link"
              label="Cancel"
              onClick={[Function]}
              size="medium"
              testId="cancel"
              type="secondary back"
            />
            <SubmitButton
              customClass="btn btn_primary cart_button btn_action"
              testId="continue"
              value="Continue"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
  <SwagLabsFooter />
</div>
`;
